LaunchPadBooster/Patching/HarmonyExtensions.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using HarmonyLib;
+using Debug = UnityEngine.Debug;
+
+namespace LaunchPadBooster.Patching
+{
+    public static class HarmonyExtensions
+    {
+        /** Runs a conditional patch on all types in the calling assembly.
+         * Note: Will run a patch on all classes in the assembly.
+         * If you wish to patch on a specific class,
+         * use <see cref="ConditionalPatchAll(Harmony, Type)"/>
+         * or <see cref="ConditionalPatch(Harmony, Type)"/> 
+         * <param name="harmony">Harmony Object</param>
+         */
+        public static void ConditionalPatchAll(this Harmony harmony)
+        {
+            AccessTools.GetTypesFromAssembly(new StackTrace().GetFrame(1).GetMethod().ReflectedType?.Assembly).Do(type => ConditionalPatch(harmony, type));
+        }
+
+        /** Runs a recursive conditional patch on all types nested in the given type
+         * Inclusive of the parent type
+         * <param name="harmony">Harmony Object</param>
+         * <param name="type">Specified type</param>
+         */
+        public static void ConditionalPatchAll(this Harmony harmony, Type type)
+        {
+            type.GetNestedTypes().Do(nType => ConditionalPatchAll(harmony, nType));
+            ConditionalPatch(harmony, type);
+        }
+        
+        /** Runs a conditional patch on the specific type referenced.
+         *  Notice: Does not apply patches in child classes.
+         *  Use <see cref="ConditionalPatchAll(Harmony, Type)"/> for this use case
+         * <param name="harmony">Harmony Object</param>
+         * <param name="type">Specified type</param>
+         */
+        public static void ConditionalPatch(this Harmony harmony, Type type)
+        {
+            var patch = type.GetCustomAttributes(true).OfType<HarmonyConditionalPatch>().FirstOrDefault();
+                
+            if (patch != null && !patch.CanPatch())
+            {
+                Debug.Log($"Patch class {type.FullName} ignored because specified condition is false!");
+                Debug.Log(patch.Description);
+                return;
+            }
+            new HarmonyConditionalClassProcessor(harmony, type).Patch();
```

Comment:
Since we're leaving the default `allowUnannotatedType=false` parameter here, if any of the `HarmonyConditionalPatch` attributes are the only ones on a class, the class will be skipped. Do we want to require another harmony attribute for these to get picked up?

Reply:
are you thinking of a case where we have a nested class with no harmony attributes on the parent but harmony attributes on the child?  From what I have gathered with how the class processor works, it only does the operation on the methods in a class and does not look for any child classes.  Child classes are loaded individually via the getalltypesfromassembly method.

I am not sure the best approach yet and will have to think about it.

Reply:
took another look at it.  I think it is fine for them to be skipped since the assembly list will list all inner classes and with inherit, if there is an attribute on the class above a method, it will get injected into the method attribute call.  I will update the logic above though to loop through all candidates instead of picking the first match.